core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractServiceExtension.java

```diff
@@ -74,38 +81,18 @@ private void registerServices(ServiceExtensionContext context) {
         var agentService = new ParticipantAgentServiceImpl();
```

Comment:
does it really make sense to have a null asset index? In case the `AssetIndex` is null the extension should throw an exception IMO.

Reply:
I agree but not for this PR. That was stuff that was already there.